app/uk/gov/hmrc/economiccrimelevyreturns/controllers/AmlRegulatedActivityController.scala

```diff
@@ -53,19 +58,42 @@ class AmlRegulatedActivityController @Inject() (
   }
 
   def onSubmit(mode: Mode): Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
+    implicit val hc: HeaderCarrier = CorrelationIdHelper.getOrCreateCorrelationId(request)
     form
       .bindFromRequest()
       .fold(
         formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, request.startAmendUrl))),
-        amlRegulatedActivity =>
-          eclReturnsConnector
-            .upsertReturn(
-              dataCleanup
-                .cleanup(request.eclReturn.copy(carriedOutAmlRegulatedActivityForFullFy = Some(amlRegulatedActivity)))
+        carriedOutAmlRegulatedActivityForFullFy => {
+          val eclReturn =
+            dataCleanup.cleanup(
+              request.eclReturn.copy(
+                carriedOutAmlRegulatedActivityForFullFy = Some(carriedOutAmlRegulatedActivityForFullFy)
```

Comment:
If you look up, there seems to be a disparity between the value here and what was changed. Shouldn't it be amlRegulatedActivity?

Reply:
I'm pretty sure we've just renamed this so that it's a more descriptive name